### ISSUE TITLE
clip-path not vendor prefixed 

### ISSUE THREAD

**Issue by jforaker** [styled-components/styled-components#1211]
`"styled-components": "^2.2.0",`

clip-path property is not being properly vendor prefixed.

actual code:

```
export const TopBlock = styled.div`
  clip-path: polygon(0 0, 100% 0, 100% 99%, 0 73%);
  background: #50E3C2;
  height: 160px;
  margin-top: 0 !important;
`;
```

output:
![image](https://user-images.githubusercontent.com/2042031/31251526-2215e7ac-a9ec-11e7-8a82-1bb5b9ac143a.png)

You will see the css does not include `webkit-clip-path`. This is problematic because Safari only respects the `webkit` version of the property.

**Comment by philpl** [comment]
@thysultan MDN lists this as unsupported by safari and the prefix doesn’t seem to be respected in a lot of browsers. caniuse also doesn’t list this

https://developer.mozilla.org/en-US/docs/Web/CSS/clip-path

I’m going to leave the decision whether to prefix it up to you, but I’m a little confused on this one

**Comment by jforaker** [comment]
The most compelling argument i can think of is the fact that autoprefixer adds it:

https://github.com/postcss/autoprefixer/wiki/support-list#mask

I reckon that MDN article is out of date. Because I'm using it actively in Safari with success.

**Comment by mxstbr** [comment]
/cc @thysultan we should add this based on caniuse:

![screen shot 2017-10-06 at 9 47 38 am](https://user-images.githubusercontent.com/7525670/31268178-6af65404-aa7b-11e7-8293-1f9ad61bc8a4.png)

**Comment by thysultan** [comment]
All the other mask variants are vendor prefixed so it makes sense that we should handle this as well.

Patched in 3.2.20

**Comment by mxstbr** [comment]
Thanks @thysultan!
